analyze_field_completion.R

```diff
@@ -0,0 +1,212 @@
+library(synapser)
+library(dplyr)
+library(tidyr)
+
+# Login to Synapse
+synLogin()
+
+# Read base Resource table
+cat("Reading base Resource data from Synapse...\n")
+resource_query <- synTableQuery(
+  "SELECT * FROM syn26450069"
+)
+resource_df <- as.data.frame(resource_query)
+
+# Read type-specific tables
+cat("Reading Animal Model data...\n")
+animal_model_query <- synTableQuery("SELECT * FROM syn26486808")
+animal_model_df <- as.data.frame(animal_model_query)
+
+cat("Reading Antibody data...\n")
+antibody_query <- synTableQuery("SELECT * FROM syn26486811")
+antibody_df <- as.data.frame(antibody_query)
+
+cat("Reading Biobank data...\n")
+biobank_query <- synTableQuery("SELECT * FROM syn26486821")
+biobank_df <- as.data.frame(biobank_query)
+
+cat("Reading Cell Line data...\n")
+cell_line_query <- synTableQuery("SELECT * FROM syn26486823")
+cell_line_df <- as.data.frame(cell_line_query)
+
+cat("Reading Genetic Reagent data...\n")
+genetic_reagent_query <- synTableQuery("SELECT * FROM syn26486832")
+genetic_reagent_df <- as.data.frame(genetic_reagent_query)
+
+# Join base resource data with type-specific data
+cat("Joining resource data with type-specific tables...\n")
+
+# Create full dataset by joining based on type-specific IDs
+resource_df <- resource_df %>%
```

Comment:
You might be able to just use this Materialized View (https://www.synapse.org/Synapse:syn51730943/tables/) for this rather than do all the joins yourself :) - not 100% sure that it has all the data you need though